configure.ac

```diff
@@ -251,28 +251,34 @@ AC_TYPE_INT16_T
 AC_TYPE_INT32_T
 AC_TYPE_INT64_T
 
+PKG_CHECK_MODULES([libqb], [libqb])
+
 if test "x$enable_man" = "xyes"; then
 	AC_ARG_VAR([DOXYGEN], [override doxygen executable])
 	AC_CHECK_PROGS([DOXYGEN], [doxygen], [no])
 	if test "x$DOXYGEN" = xno; then
 		AC_MSG_ERROR(["Doxygen command not found"])
 	fi
-	# required by doxyxml to build man pages dynamically
-	# Don't let AC_PROC_CC (invoked by AX_PROG_CC_FOR_BUILD) replace
-	# undefined CFLAGS_FOR_BUILD with -g -O2, overriding our special OPT_CFLAGS.
-	: ${CFLAGS_FOR_BUILD=""}
-	AX_PROG_CC_FOR_BUILD
+
+	AC_ARG_VAR([DOXYGEN2MAN], [override doxygen2man executable])
+
+	# required to detect doxygen2man when libqb is installed
+	# in non standard paths
 	saved_PKG_CONFIG="$PKG_CONFIG"
 	saved_ac_cv_path_PKG_CONFIG="$ac_cv_path_PKG_CONFIG"
 	unset PKG_CONFIG ac_cv_path_PKG_CONFIG
 	AC_PATH_PROG([PKG_CONFIG], [pkg-config])
 	PKG_CHECK_MODULES([libqb_BUILD], [libqb])
-	PKG_CHECK_MODULES([libxml_BUILD], [libxml-2.0])
+	PKG_CHECK_VAR([libqb_BUILD_PREFIX], [libqb], [prefix])
+	AC_PATH_PROG([DOXYGEN2MAN], [doxygen2man], [no], [$libqb_BUILD_PREFIX/bin$PATH_SEPARATOR$PATH])
```

Comment:
Isn't it too unexpected to prefer a custom-installed `doxygen2man` above one in `$PATH`?

Reply:
actually I have done this on purpose.

Say you have:
apt-get install libqb-dev doxygen2man
and you have a manually installed copy in /srv/foo/bar

and for the build you want to use the non-standard copy:
export PKG_CONFIG_PATH=/srv/foo/bar/lib/pkgconfig
./configure

you want to use a version of doxygen2man associated to the requested version of libqb and not the system path.

Of course we can override the variable at configure time, but I think it´s more elegant if we can autodetect it correctly.

Reply:
I can certainly accept this, though in my head libqb and doxygen2man aren't related. Your (as of now) hypothetical `apt-get` line suggests so as well, because it isn't visible that they're built from a single source tree. Your "insider" view can well be different, of course. Since it's easy to override either way, I'm fine with the choice which is more convenient to you.

Reply:
@wferi the apt was to show that you can have multiple version of library and binary installed on the system and we want to prefer the binary from the alternative installation when that is chosen to build the docs.
I agree that libqb and doxygen2man are supposed to be somewhat disconnected, and from a packager perspective it makes all sense, but for me, building from source, using all but packages, it´s one less variable to pass around each time I build stuff.
Anyway I am glad it´s not going to be a big issue and we are both happy with the solution :)

It´s always great to work with you :)

Cheers
Fabio